osh/word_eval.py

```diff
@@ -61,6 +61,26 @@
 # For compatibility, ${BASH_SOURCE} and ${BASH_SOURCE[@]} are both valid.
 # ${FUNCNAME} and ${BASH_LINENO} are also the same type of of special variables.
 _STRING_AND_ARRAY = ['BASH_SOURCE', 'FUNCNAME', 'BASH_LINENO']
+def CheckCompatArray(var_name, opts, compat_introspect=True):
```

Comment:
I would name `compat_introspect` something more descriptive like `is_plain_var` or `is_plain_var_sub`

( The difference between SimpleVarSub and BracedVarSub is one thing I want to get rid of with #604 )

Reply:
I renamed it to `is_plain_var_sub`.